icu4c/source/i18n/number_longnames.cpp

```diff
@@ -778,6 +779,107 @@ const UChar *trimSpaceChars(const UChar *s, int32_t &length) {
     return s + start;
 }
 
+UnicodeString calculateGenderForUnit(const Locale &locale, const MeasureUnit &unit, UErrorCode &status) {
```

Comment:
Move the comment discussing the purpose of this function, and how it relates to forArbitraryUnit/processPatternTimes, to a doc comment, and elaborate a bit more.